### PR TITLE
Add translations for submenu tray back button

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-spectrum/menu/intl/ar-AE.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
   "moreActions": "المزيد من الإجراءات",
-  "unavailable": "غير مُتوفر، قُم بالتوسيع للحصول على التفاصيل"
+  "unavailable": "غير مُتوفر، قُم بالتوسيع للحصول على التفاصيل",
+  "backButton": "Return to "
```
Comment (LFDanLu) [inline packages/@react-spectrum/menu/intl/ar-AE.json:4]: SubMenu tray back button has the above as its aria-label and then is aria-labelledby a neighboring span that contains the text of its submenu trigger. See build attached to [66ce969](https://github.com/adobe/react-spectrum/pull/4976/commits/66ce969411c91d333a86eb901ac1f69eabda7f3d) (will add link when build finishes

Comment (reidbarber) [reply packages/@react-spectrum/menu/intl/ar-AE.json:4]: Is there a way we can revert back to this? (without getting double announcements)

```suggestion
  "backButton": "Return to {prevMenuButton}"
```

Not sure the relying on the aria-labelledby will make sense in some locales in terms of ordering.

Comment (LFDanLu) [reply packages/@react-spectrum/menu/intl/ar-AE.json:4]: I don't quite remember if the ordering here is actually a problem in other locales, do we have an example? I noticed a bunch of other components use the same labeling in LTR as RTL (aka useColorArea, etc)

Comment (reidbarber) [reply packages/@react-spectrum/menu/intl/ar-AE.json:4]: One example is in dnd:

en-US: `"dragItem": "Drag {itemText}"`
de-DE: `"dragItem": "{itemText} ziehen"`

and

en-US: `"dropOnItem": "Drop on {itemText}"`
de-DE: `"dropOnItem": "Auf {itemText} ablegen"`.

Comment (LFDanLu) [reply packages/@react-spectrum/menu/intl/ar-AE.json:4]: Sweet thanks, I can def revert to my old approach then

**Review by yihuiliao — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**